src/features/wisdom/wisdomOAuthProvider.ts

```diff
@@ -0,0 +1,458 @@
+   * @returns
+   */
+  public async createSession(scopes: string[]): Promise<AuthenticationSession> {
+    try {
+      const account = await this.login(scopes);
+
+      if (!account) {
+        throw new Error(`Wisdom login failure`);
+      }
+
+      // const userinfo: { name: string; email: string } = await this.getUserInfo(
+      //   token
+      // );
+
+      const identifier = uuid();
+      const session: AuthenticationSession = {
+        id: identifier,
+        accessToken: account.accessToken,
+        account: {
+          label: "test-common",
+          id: identifier,
+        },
+        scopes: [],
+        // scopes: account.scope,
+      };
+
+      await this.context.secrets.store(
+        SESSIONS_SECRET_KEY,
+        JSON.stringify([session])
+      );
+
+      this._sessionChangeEmitter.fire({
+        added: [session],
+        removed: [],
+        changed: [],
+      });
+
+      console.log("[oauth] Session created...");
+
+      return session;
+    } catch (e) {
+      window.showErrorMessage(`Sign in failed: ${e}`);
+      throw e;
+    }
+  }
+
+  /**
+   * Remove an existing session
+   * @param sessionId
+   */
+  public async removeSession(sessionId: string): Promise<void> {
+    const allSessions = await this.context.secrets.get(SESSIONS_SECRET_KEY);
+    if (allSessions) {
+      const sessions = JSON.parse(allSessions) as AuthenticationSession[];
+      const sessionIdx = sessions.findIndex((s) => s.id === sessionId);
+      const session = sessions[sessionIdx];
+      sessions.splice(sessionIdx, 1);
+
+      await this.context.secrets.store(
+        SESSIONS_SECRET_KEY,
+        JSON.stringify(sessions)
+      );
+
+      if (session) {
+        this._sessionChangeEmitter.fire({
+          added: [],
+          removed: [session],
+          changed: [],
+        });
+        window.registerTreeDataProvider(
+          "wisdom",
+          new TreeDataProvider(undefined)
+        );
+      }
+    }
+  }
+
+  /**
+   * Dispose the registered services
+   */
+  public async dispose() {
+    this._disposable.dispose();
+  }
+
+  /* Log in to wisdom auth service*/
+  private async login(scopes: string[] = []) {
+    console.log("[oauth] Logging in...");
+
+    const searchParams = new URLSearchParams([
+      ["response_type", "code"],
+      ["code_challenge", CODE_CHALLENGE],
+      ["code_challenge_method", "S256"],
+      ["client_id", CLIENT_ID],
+      ["redirect_uri", this.redirectUri],
+    ]);
+    const uri = Uri.parse(
+      `${OAUTH_BASE_PATH}/o/authorize?${searchParams.toString()}`
+    );
+    console.log("[oauth] uri -> ", uri.toString());
+
+    const {
+      promise: receivedRedirectUrl,
+      cancel: cancelWaitingForRedirectUrl,
+    } = promiseFromEvent(this._uriHandler.event, this.handleUriForCode(scopes));
+
+    await env.openExternal(uri);
+
+    const account = await window.withProgress(
+      {
+        title: "Waiting for OAuth redirect from Wisdom Service",
```

Comment:
s/Waiting for OAuth redirect from Wisdom Service/Waiting for authentication redirect from Ansible wisdom Service